room-ease/src/containers/UserProfile/UserProfile.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import RegularButton from "../../components/inputs/RegularButton";
+import TextInput from "../../components/inputs/TextInput";
+import { Link, withRouter } from "react-router-dom";
+
+import * as classes from './userprofile.module.css';
+import axios from 'axios';
+import {MEMBERS_OF_ROOM_URL,USER_PROFILE_URL} from '../../constants/ServerRoutes';
+import { withLayout } from '../../hoc/Layout/withLayout'
+
+class UserProfile extends Component{
+    constructor(props){
+    super(props);
+        this.state={
+            roomInfo:[],
+            userInfo:{},
+        }
+    }
+    
+
+    componentDidMount(){
+        this.hasRoom()
+    }
+
+    hasRoom= async ()=>{
+        let userID="5f7918453c2bc625c03ecb20"
+        let roomName=localStorage.getItem('roomName')
+        console.log("adasd",userID,roomName)
+        let profileStatus= (await axios.get(USER_PROFILE_URL+"?userID="+userID+"&roomName="+roomName)).data;
+        let {Result,Error}= profileStatus
+        console.log("received data",profileStatus.userInfo)
+        this.setState({roomInfo:profileStatus.Members});
+        let obj={}
+        obj["Username"]=profileStatus.userInfo.userName;
```

Comment:
Getting this error Cannot read property 'userName' of undefined Handle using try catch